packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -270,15 +277,47 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
     // Only remove self from the GroupCall. Contoso must manage access to Chat.
     await this.callAdapter.leaveCall(forEveryone);
   }
+
+  /** Start a new Call. */
+  public startCall(participants: string[], options?: StartCallOptions): Call | undefined;
   /** Start a new Call. */
   public startCall(
-    participants: string[] | /* @conditional-compile-remove(PSTN-calls) */ CommunicationIdentifier[],
+    participants: (
+      | MicrosoftTeamsAppIdentifier
+      | /* @conditional-compile-remove(PSTN-calls) */ PhoneNumberIdentifier
+      | /* @conditional-compile-remove(one-to-n-calling) */ CommunicationUserIdentifier
+      | /* @conditional-compile-remove(teams-adhoc-call) */ MicrosoftTeamsUserIdentifier
+      | UnknownIdentifier
+    )[],
+    options?: StartCallOptions
+  ): Call | undefined;
+  /** Start a new Call. */
+  public startCall(
+    participants:
+      | string[]
+      | (
+          | MicrosoftTeamsAppIdentifier
+          | /* @conditional-compile-remove(PSTN-calls) */ PhoneNumberIdentifier
+          | /* @conditional-compile-remove(one-to-n-calling) */ CommunicationUserIdentifier
+          | /* @conditional-compile-remove(teams-adhoc-call) */ MicrosoftTeamsUserIdentifier
+          | UnknownIdentifier
+        )[],
     options?: StartCallOptions
   ): Call | undefined {
-    let communicationParticipants = participants;
-    /* @conditional-compile-remove(PSTN-calls) */
-    communicationParticipants = participants.map(_toCommunicationIdentifier);
-    return this.callAdapter.startCall(communicationParticipants, options);
+    if (typeof participants[0] === 'string') {
```

Comment:
Not sure of the purpose of this `if`. Both the if and the else do the exact same functionality:
`return this.callAdapter.startCall(participants, options);`

We shouldn't use runtime logic for typing (especially as we are casting with an `as` below anyway)

Reply:
Since we are passing this to another function with overloads we can't pass it the implementation signiture, we need to make sure it is cast since it has to be one of the two overloads on the new objects function. (since we are passing to the CallAdapter here.)

Reply:
Ah yeah typescript doesn't support overloads that well as it would be too complex to make the typing work
https://github.com/microsoft/TypeScript/issues/12041